### PR TITLE
Remove voltage level and its equipments

### PR DESCRIPTION
**Pull request by etiennehomer**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature: this PR implements the 'remove voltage level' modification. When the modification is applied, the voltage level, its equipments and the branches it is connected to are removed.

### CODE REVIEW

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveVoltageLevel.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(RemoveVoltageLevel.class);
+
+    private final String voltageLevelId;
+
+    public RemoveVoltageLevel(String voltageLevelId) {
+        this.voltageLevelId = Objects.requireNonNull(voltageLevelId);
+    }
+
+    @Override
+    public void apply(Network network, boolean throwException, ComputationManager computationManager, Reporter reporter) {
```
Comment (annetill) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveVoltageLevel.java:34]: To everybody:
- This feature (but we have this issue also for a classical connectable remove) does not handle extensions? Or it is hidden in the remove of each connectable? A `VoltageLevel` can also have extensions. What is our position on that?
- I have an issue with the remove of connectables in general, that we do here. In some network, we are using `regulatingTerminal` that can be remote. So I think that we can remove a connectable which terminal can be used for a remote regulation.

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveVoltageLevel.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
```
Comment (miovd) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveVoltageLevel.java:6]: Can you add the SPDX? (example :
```/**
 * Copyright (c) 2023, RTE (http://www.rte-france.com/)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 * SPDX-License-Identifier: MPL-2.0
 */)

File: iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/topology/RemoveVoltageLevelTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author Etienne Homer <etienne.homer at rte-france.com>
+ */
+public class RemoveVoltageLevelTest {
```
Comment (miovd) [inline iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/topology/RemoveVoltageLevelTest.java:26]: I think you can add a round trip test on an xml test just to test the rest of the network has not been touched by the modification

**Review by miovd — CHANGES_REQUESTED** [review]

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveVoltageLevel.java (hunk 1 above)
Comment (miovd) [reply iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveVoltageLevel.java:34]: About the first point, it is already handled by the `remove` method!

About the second point: it is indeed an issue, to be discussed in another PR about removal in general.

File: iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/topology/RemoveVoltageLevelTest.java (hunk 3 above)
Comment (etiennehomer) [reply iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/topology/RemoveVoltageLevelTest.java]: I added a round trip test for node breaker and bus breaker topology. Does it make sense to test both ?

**Review by miovd — APPROVED** [review]

**Review by geofjamg — APPROVED** [review]

**Pull request merged**